src/PETScNonlinearSolvers.jl

```diff
@@ -171,16 +159,16 @@ function Algebra.solve!(x::T,
                         cache::PETScNonlinearSolverCache{<:T}) where T <: AbstractVector
 
   @assert cache.op === op
-  @check_error_code PETSC.SNESSolve(nls.snes[],C_NULL,cache.x_petsc_vec.vec[])
+  @check_error_code PETSC.SNESSolve(cache.snes[],C_NULL,cache.x_petsc_vec.vec[])
   _copy_and_exchange!(x,cache.x_petsc_vec)
   cache
 end
 
-function Algebra.solve!(x::AbstractVector,nls::PETScNonlinearSolver,op::NonlinearOperator)
+function Algebra.solve!(x::AbstractVector,nls::PETScNonlinearSolver,op::NonlinearOperator,::Nothing)
   cache=_setup_cache(x,nls,op)
   _set_petsc_residual_function!(nls,cache)
```

Comment:
Just an stylistic (totally subjective and optional) remark:

Functions `_set_petsc_residual_function!`,  `_set_petsc_jacobian_function!`, and `_setup_cache` are very short and are used only once.  Removing them and adding the code inline would improve readability IMO.

Reply:
Done https://github.com/gridap/GridapPETSc.jl/pull/53/commits/0379aa105f471bf85d68aeb9ec03b4e174de92f8